File: services/python/device-location-aggregator/models/beacon.py
import time
from dataclasses import dataclass
from typing import Dict


class Beacon:
    ID: str
    rssi: int
    lastUpdated: int

    def __init__(self, ID: str, rssi: int):
        self.ID = ID
        self.rssi = rssi
        self.lastUpdated = int(time.time())

    def UpdateBeacon(self, rssi: int):
        self.rssi = rssi
        self.lastUpdated = int(time.time())

    def to_dict(self) -> Dict:
        return {
            "ID": self.ID,
            "rssi": self.rssi,
            "lastUpdated": self.lastUpdated
        }

File: ui/src/app/models/beacon.ts

export type BeaconId = string

// models/beacon.model.ts
export class Beacon {
  constructor(
    public ID: BeaconId,
    public rssi: number,
    public lastUpdated: number
  ) {}

  updateBeacon(rssi: number): void {
    this.rssi = rssi;
    this.lastUpdated = Date.now();
  }
}
